source/cgal/surface_mesh.cc

```diff
@@ -114,6 +116,105 @@ namespace CGALWrappers
                     (f % 2 == 0));
   }
 
+
+
+  template <typename CGALPointType, int dim, int spacedim>
+  void
+  boolean_operation(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell1,
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell2,
+    CGAL::Surface_mesh<CGALPointType> &                         outsm,
+    const BooleanOperation &boolean_operation)
+  {
+    Assert(dim != 1 || spacedim != 1,
+           ExcMessage(
+             "This function does not work with 1-dimensional objects."));
+    Assert(
+      outsm.is_empty(),
+      ExcMessage(
+        "The output surface mesh needs to be empty upon calling this function."));
+    namespace PMP = CGAL::Polygon_mesh_processing;
+    CGAL::Surface_mesh<CGALPointType> sm1, sm2;
+    dealii_cell_to_cgal_surface_mesh(cell1,
+                                     MappingQ<dim, spacedim>(1),
+                                     sm1); //[TODO: remove Mapping from here...]
+    dealii_cell_to_cgal_surface_mesh(cell2, MappingQ<dim, spacedim>(1), sm2);
+    PMP::triangulate_faces(sm1);
+    PMP::triangulate_faces(sm2);
+
+    [[maybe_unused]] bool res = false;
+    switch (boolean_operation)
+      {
+        case BooleanOperation::UNION:
+          res = PMP::corefine_and_compute_union(sm1, sm2, outsm);
+          break;
+        case BooleanOperation::INTERSECTION:
+          res = PMP::corefine_and_compute_intersection(sm1, sm2, outsm);
+          break;
+        case BooleanOperation::DIFFERENCE:
+          res = PMP::corefine_and_compute_difference(sm1, sm2, outsm);
+          break;
+        case BooleanOperation::NONE:
+          PMP::corefine(sm1, sm2);
+          (void)outsm;
+          res = true;
+          break;
+        default:
+          Assert(
+            res,
+            ExcMessage(
+              "The boolean operation you provided doesn't make sense. Please check it."));
+          break;
+      }
+    Assert(res,
+           ExcMessage("The boolean operation was not succesfully computed."));
+  }
+
+
+
+  template <typename CGALPointType, typename CGALTriangulation>
+  Quadrature<CGALPointType::Ambient_dimension::value>
+  collect_quadratures_inside_surface(
```

Comment:
maybe a name like `build_quadrature_on_convex_hull`? 

Also, remove GCALTriangulation from the input argument. Since you are only using it inside the function, just build it inside.

Reply:
We should also provide a different version of this function for non-convex polyhedral meshes. Maybe adding a boolean argument like `integrate_on_convex_hull` which defaults to `false` that uses the algorithm you showed me yesterday, i.e., create a domain, then a mesh, then last lines.